synapse/lib/cell.py

```diff
@@ -1648,6 +1655,11 @@ async def configNexsVers(self):
                 ('setUserProfInfo', self._setUserProfInfoV0),
             ])
 
+        if self.nexsvers < (2, 197):
+            await self._bumpCellVers('cell:storage', (
+                (3, self._storCellAuthLockArchivedUsers),
```

Comment:
This doesn't seem quite right, it runs the migration when configNexsVers is called and we aren't at 2.197?

Reply:
Yeah, when `configNexsVers()` is called, `self.nexsvers` is the current nexus version instead of the version we're moving to.

Reply:
So, for example, when moving from 2.177 to 2.197 `setNexsVers((2, 197))` is called which calls `configNexsVers()` when `self.nexsvers` is currently `(2, 177)` and _then_ sets `self.nexsvers` to `(2, 197)`. Hopefully that makes sense.

Reply:
configNexsVers is also called while the cell is starting up though so isn't it going to migrate on startup?

Reply:
Nope, because the migration is dependent on the value of `self.nexsvers` which will still be the old version until the cell becomes active and calls `setNexsVers()`